src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -341,10 +343,13 @@ trait MethodSynthesis {
         case (p :: Nil) :: _  => p
         case _                => NoSymbol
       }
-      private def setterRhs = (
-        if (mods.isDeferred || derivedSym.isOverloaded) EmptyTree
+      // TODO: when is `derivedSym.isOverloaded`??? is it always an error?
```

Comment:
I don't see how we'd get here. In erroneous code like `class C(var x: Any) { var x: Any }` we issue the double def error during namer, and don't enter the second symbol in scope. So when this code is called (during typer), we don't end up with an overloaded setter.

Furthermore, we can't inherit overloads, as `DerivedSetter#derivedSym` uses `getterIn(enclClass)`, which only looks at `decls`.

I tried turning it into an assertion, but didn't trip it in `partest --neg --pos`.